### ISSUE TITLE
Needs a proper way to reorder on the Z-index

### ISSUE THREAD

**Issue by JellyWX** [kivy/kivy#5450]
Kivy currently has no proper way to reorder elements on the Z-index. The only way is by sorting the child list and then using `remove_widget(i); add_widget(i)`. This is not ideal for most situations, since it means that for an instant the widget's parent becomes `NoneType` and the program crashes if you're doing positioning relative to parent size.
TL;DR: Please add a proper way to order element on the z-index, possibly just a `.to_front()` would work nicely.

**Comment by tito** [comment]
When you talk about z-index, it's the index of the widget in the its parent children list right?

**Comment by JellyWX** [comment]
yes, basically. however, simply reordering the children list doesnt work.

**Comment by kuguma** [comment]
here is workaround:

`
class Widget(WidgetBase):
...
    def to_front(self, widget):
        target = None
        if widget.canvas in self.canvas.children:
            target = self.canvas.children
        elif widget.canvas in self.canvas.after.children:
            target = self.canvas.after.children
        elif widget.canvas in self.canvas.before.children:
            target = self.canvas.before.children
        else:
            return
        top = len(target)-1
        i = target.index(widget.canvas)
        target[top],target[i] = target[i],target[top]
...
`

**Comment by JellyWX** [comment]
looks very neat, havent got a set up to properly test it however. would be nice if it could somehow fit into a PR maybe, but good for future reference. Thank you

**Comment by Nasjoe** [comment]
Hi ! Can you give us an example with two widgets, like an image ? Many Thanks !

**Comment by JellyWX** [comment]
https://github.com/JellyWX/sample-ordering-kivy

sorry for slow reply!

⁣Get BlueMail for Android ​
